orca-core/src/test/groovy/com/netflix/spinnaker/orca/pipeline/util/BuildDetailExtractorSpec.groovy

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright 2016 Schibsted ASA.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.netflix.spinnaker.orca.pipeline.util
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import org.springframework.beans.factory.annotation.Autowired
+import spock.lang.Shared
+import spock.lang.Specification
+import spock.lang.Unroll
+
+class BuildDetailExtractorSpec extends Specification {
+
+  @Autowired
+  ObjectMapper mapper
+
+  @Shared
+  BuildDetailExtractor buildDetailExtractor = new BuildDetailExtractor()
+
+  @Unroll
+  def "Default detail from buildInfo"() {
+
+    when:
+    buildDetailExtractor.tryToExtractBuildDetails(buildInfo, request)
+
+    then:
+    request == result
+
+    where:
+    buildInfo                                                                                                                                              | request | result
+
```

Comment:
Reformated.

About the 
result = request + buildInfo
It will never be true, since the result is the expectedResult after parsing the buildInfo. The parsing is stored in the request object.